### PR TITLE
Replacing observed OTUs with richness

### PR DESCRIPTION
**Pull request by qiyunzhu**
Please complete the following checklist:

The metric `observed_otus` may be outdated, and was replaced by `observed_features` in QIIME 2. However it is still `observed_otus` in scikit-bio. I think that the most accurate term in community ecology is "observed species richness". It is often denoted as $S_{obs}$ or $S$ in literature.

Therefore I am suggesting a new metric `sobs` to indicate "observed species richness". The word "sobs", although brief, is consistent with the naming convention of current alpha diversity metrics, which usually consists of human names, mathematical symbols and abbreviations (such as `simpson_d`, `fisher_alpha`, `ace`). Whereas a name that describes the nature of the metric may be too long. For example, `observed_richness` is a good descriptive name, but may not be ideal since the metric is the simplest one of all.

For reference: In vegan, this metric can be calculated using the function `specnumber` (meaning species number), and it is denoted as `S.obs` in the output of `estimateR`.

Additionally, I added `observed_features` to be consistent with QIIME 2, and deprecated `observed_otus`. In this PR, the three metrics: `observed_otus`, `observed_features` and `sobs` are aliases.

***

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/.github/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by qiyunzhu** [comment]
@mortonjt Would you be interested in taking a look at this PR? Thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### Features
 
+* Added alpha diversity metric `sobs`, which is the observed species richness (S_{obs}) of a sample. It means to replace `observed_otus` which uses the historical term "OTU". Also added metric `observed_features` to be compatible with the QIIME 2 terminology. All three metrics are equivalent ([#1902](https://github.com/biocore/scikit-bio/pull/1902)).
```
Comment (mataton) [inline CHANGELOG.md:11]: @qiyunzhu The sentence `It means to replace 'observed_otus' ...` could be clearer if we said `'sobs' will replace 'observed_otus', ...`, or `'sobs' replaces 'observed_otus', ...`. Also, note the comma before the word `which`.

**Comment by mataton** [comment]
@qiyunzhu Other than my comment on the grammar used in `CHANGELOG.md`, everything looks good. I built the documents locally and read through the code several times, everything renders correctly.

File: CHANGELOG.md (hunk 1 above)
Comment (qiyunzhu) [reply CHANGELOG.md:11]: Thanks @mataton I guess you can suggest changes in the GitHub web interface. Would like to give it a try?

**Pull request merged**